core/domain/event_services.py

```diff
@@ -133,12 +133,29 @@ class CompleteExplorationEventHandler(BaseEventHandler):
 
     @classmethod
     def _handle_event(
-            cls, exp_id, exp_version, state_name, session_id, time_spent,
-            params, play_type):
+            cls, exp_id, exp_version, state_name, collection_id, user_id,
+            session_id, time_spent, params, play_type):
         stats_models.CompleteExplorationEventLogEntryModel.create(
             exp_id, exp_version, state_name, session_id, time_spent,
             params, play_type)
 
+        if user_id and collection_id:
+            exp_completion_model = (
+                user_models.ExplorationInCollectionCompletionModel.get_or_create(
+                    user_id, collection_id))
+
+            completed_explorations = set()
```

Comment:
Not exactly, it was a set to guard against an exploration being played a second time and ensuring that ID was not duplicated inside of the context. I just noticed looking over this code again that there was a conditional already guarding against this, so it's been changed to a list.